[BUG]: DOCA build failure on ARM

### Version

25.02

### Which installation method(s) does this occur on?

Docker

### Describe the bug.

```
-- [morpheus.python.morpheus.morpheus._lib.libmd.rapids.cpm.find] CPM: Adding package md@1.0.4 (1.0.4 to /Morpheus/.cache/cpm/md/5c907728cd0baebae9b9f7496806e4d7338ed05a)
-- [morpheus.python.morpheus.morpheus._lib.libmd] libmd was not installed and will be built from source
-- [morpheus.python.morpheus.morpheus._lib.libbsd.rapids.cpm.find] CPM: Adding package bsd@0.11.7 (0.11.7 to /Morpheus/.cache/cpm/bsd/120b2246daf5c47f371af3f83399e8ed828b56d8)
-- [morpheus.python.morpheus.morpheus._lib.libbsd] libbsd was not installed and will be built from source
-- [morpheus.python.morpheus.morpheus._lib] Could NOT find libdpdk (missing: rte_eal_LIBRARY) 
CMake Error at /opt/conda/envs/morpheus/share/cmake-3.27/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find doca (missing: libdpdk_FOUND doca_eth_LIBRARY
  doca_flow_LIBRARY doca_gpunetio_LIBRARY doca_gpunetio_device_LIBRARY)
  (found version "")
Call Stack (most recent call first):
  /opt/conda/envs/morpheus/share/cmake-3.27/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  cmake/package_search/Finddoca.cmake:110 (find_package_handle_standard_args)
  python/morpheus/morpheus/_lib/doca/CMakeLists.txt:20 (find_package)
```

### Minimum reproducible example

```shell

```

### Relevant log output

<details><summary>Click here to see error details</summary><pre>

 [Paste the error here, it will be hidden by default]

</pre></details>


### Full env printout

<details><summary>Click here to see environment details</summary><pre>

 [Paste the results of print_env.sh here, it will be hidden by default]

</pre></details>


### Other/Misc.

_No response_

### Code of Conduct

- [x] I agree to follow Morpheus' Code of Conduct
- [x] I have searched the [open bugs](https://github.com/nv-morpheus/Morpheus/issues?q=is%3Aopen+is%3Aissue+label%3Abug) and have found no duplicates for this bug report

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -132,6 +132,13 @@ add_custom_target(${PROJECT_NAME}_style_checks
   COMMENT "Building dependencies for style checks"
 )
 
+# CMAKE_LIBRARY_ARCHITECTURE is needed to find DOCA dependencies
+if(NOT DEFINED CMAKE_LIBRARY_ARCHITECTURE)
+  set(CMAKE_LIBRARY_ARCHITECTURE "${CMAKE_SYSTEM_PROCESSOR}-${CMAKE_SYSTEM_NAME}-${CMAKE_CXX_COMPILER_ID}")
+  string(TOLOWER "${CMAKE_LIBRARY_ARCHITECTURE}" CMAKE_LIBRARY_ARCHITECTURE)
+  message(STATUS "Setting CMAKE_LIBRARY_ARCHITECTURE to ${CMAKE_LIBRARY_ARCHITECTURE}")
+endif()
+
 # Configure all dependencies
 include(dependencies)
 

--- a/ci/scripts/github/cmake_all.sh
+++ b/ci/scripts/github/cmake_all.sh
@@ -34,12 +34,6 @@ fi
 export CMAKE_BUILD_ALL_FEATURES="${_FLAGS[@]}"
 unset _FLAGS
 
-if [[ ${REAL_ARCH} == "aarch64" ]]; then
-    # Currently DOCA is failing to build on ARM
-    # https://github.com/nv-morpheus/Morpheus/issues/2092
-    export MORPHEUS_SUPPORT_DOCA=OFF
-fi
-
-if [[ ${MORPHEUS_SUPPORT_DOCA} == @(TRUE|ON) && ${REAL_ARCH} == "x86_64" ]]; then
+if [[ ${MORPHEUS_SUPPORT_DOCA} == @(TRUE|ON) ]]; then
     export CMAKE_BUILD_ALL_FEATURES="${CMAKE_BUILD_ALL_FEATURES} -DMORPHEUS_SUPPORT_DOCA=ON"
 fi

--- a/cmake/package_search/Finddoca.cmake
+++ b/cmake/package_search/Finddoca.cmake
@@ -42,11 +42,6 @@ block(SCOPE_FOR VARIABLES
   set("CMAKE_FIND_ROOT_PATH_MODE_INCLUDE" BOTH)
   set("CMAKE_FIND_ROOT_PATH_MODE_LIBRARY" BOTH)
 
-  # CMAKE_LIBRARY_ARCHITECTURE needs to be set for this to work correctly. Will be restored at the end of the block
-  if(NOT DEFINED CMAKE_LIBRARY_ARCHITECTURE)
-    set(CMAKE_LIBRARY_ARCHITECTURE x86_64-linux-gnu)
-  endif()
-
   # Find the include path
   find_path(
     doca_INCLUDE_DIR doca_gpunetio.h

--- a/cmake/package_search/Findlibdpdk.cmake
+++ b/cmake/package_search/Findlibdpdk.cmake
@@ -28,11 +28,6 @@ block(SCOPE_FOR VARIABLES
   # List of required args. Start with INCLUDE_DIR since the first one is displayed
   list(APPEND libdpdk_REQUIRED_VARS bsd_FOUND libdpdk_INCLUDE_DIR)
 
-  # CMAKE_LIBRARY_ARCHITECTURE needs to be set for this to work correctly. Will be restored at the end of the block
-  if(NOT DEFINED CMAKE_LIBRARY_ARCHITECTURE)
-    set(CMAKE_LIBRARY_ARCHITECTURE x86_64-linux-gnu)
-  endif()
-
   # This library will always be installed on the host. Allow that to be searched here (Should fix this up in the future)
   set("CMAKE_FIND_ROOT_PATH_MODE_INCLUDE" BOTH)
   set("CMAKE_FIND_ROOT_PATH_MODE_LIBRARY" BOTH)
@@ -76,7 +71,7 @@ block(SCOPE_FOR VARIABLES
         set_target_properties(${library_name} PROPERTIES
           IMPORTED_LINK_INTERFACE_LANGUAGES "C"
           IMPORTED_LOCATION "${${library_name}_LIBRARY}"
-          INTERFACE_INCLUDE_DIRECTORIES "${libdpdk_INCLUDE_DIR};${libdpdk_INCLUDE_DIR}/../x86_64-linux-gnu/dpdk"
+          INTERFACE_INCLUDE_DIRECTORIES "${libdpdk_INCLUDE_DIR};${libdpdk_INCLUDE_DIR}/../${CMAKE_LIBRARY_ARCHITECTURE}/dpdk"
           INTERFACE_LINK_LIBRARIES "bsd::bsd"
         )
 

